### ISSUE TITLE
Implement PARC

### ISSUE THREAD

**Issue by jack89roberts** [alan-turing-institute/ARC-LoCoMoSeT#6]
Pairwise Annotation Representation Comparison (PARC)
Bolya et al., 2022

- https://arxiv.org/pdf/2111.06977.pdf
- https://dbolya.github.io/parc/
- https://github.com/dbolya/parc

**Comment by eddableheath** [comment]
A couple of notes from the implementation:

- The PARC paper suggests that it is robust to only having a limited number of training examples, however this is also confounded by the number of classes, the test (or probe as it's called in the paper) subset of the dataset must have some images with repeated classes for it to work, how much over lap is required I don't have a good gauge for yet. More discussion of this to be found in the notebook.
- The final PARC score is a Spearman correlation, I think this means that the higher the correlation the better, but I'm currently unsure.

**Comment by eddableheath** [comment]
Having thought more about the conceptual explanation of PARC thismorning (a notion missing from the paper) I have settled on the following description for now:

> Overall PARC gives a measure suggesting how well separated the features of a model are (w.r.t. a test set of data) by giving the correlation of this to the ground truth labels, mapped to a similar space in which a metric considers the separation.

**Comment by eddableheath** [comment]
Some notes on the additional methods/considerations from the PARC paper:

- They apply dimensionality reduction (using PCA) to each set of output features to be compared as they suggest that this can be a source of miscalibration across models.
- For the CNN case the depth of the model is a strong indicator of how much information a CNN can learn, i.e. if the fine-tuning data set is is particularly difficult (they give the example of NA Birds as a difficult dataset) the depth of the model is a good proxy for how well it will learn. There might be some equivalent idea for transformers.
